main.go

```diff
@@ -176,6 +194,12 @@ func main() {
 				http.Error(w, fmt.Sprintf("failed to reload config: %s", err), http.StatusInternalServerError)
 			}
 		})
+	http.Handle("/metrics", prometheus.Handler())
+	http.HandleFunc("/probe", func(w http.ResponseWriter, r *http.Request) {
+		sc.RLock()
```

Comment:
The mutex is taken for too long here, this will block reloads while a probe is in progress. Only hold the mutex for enough time to copy over C.